e2e/accounts/account_node_test.go

```diff
@@ -0,0 +1,647 @@
+			if err != testCase.expectedError {
+				require.FailNow(fmt.Sprintf("unexpected error: expected \n'%v', got \n'%v'", testCase.expectedError, err))
+			}
+			if err == nil {
+				require.NotEmpty(addr)
+				require.NotEmpty(pubKey)
+				require.NotEmpty(mnemonic)
+			}
+		}()
+	}
+}
+
+func (s *AccountsNodeTestSuite) TestCreateChildAccount() {
+	require := s.Require()
+
+	// test cases
+	testCases := []struct {
+		context            string
+		parentAddress      string
+		parentPassword     string
+		defaultKeyAddress  string
+		defaultKeyPassword string
+		expectedError      error
+	}{
+		{
+			context:        "parent address is empty, no default account",
+			parentAddress:  "",
+			parentPassword: "password",
+			expectedError:  account.ErrNoAccountSelected,
+		},
+		{
+			context:        "parent address is not empty, incorrect address format",
+			parentAddress:  incorrectAddress,
+			parentPassword: "password",
+			expectedError:  account.ErrAddressToAccountMappingFailure,
+		},
+
+		{
+			context:        "correct parent address, incorrect password",
+			parentAddress:  TestConfig.Account2.Address,
+			parentPassword: incorrectPassword,
+			expectedError:  errors.New("cannot retrieve a valid key for a given account: could not decrypt key with given passphrase"),
+		},
+
+		{
+			context:        "correct parent address, correct password",
+			parentAddress:  TestConfig.Account2.Address,
+			parentPassword: TestConfig.Account2.Password,
+			expectedError:  nil,
+		},
+		{
+			context:            "parent address is empty, default account is not empty, incorrect password",
+			parentAddress:      "",
+			parentPassword:     incorrectPassword,
+			defaultKeyAddress:  TestConfig.Account2.Address,
+			defaultKeyPassword: TestConfig.Account2.Password,
+			expectedError:      errors.New("cannot retrieve a valid key for a given account: could not decrypt key with given passphrase"),
+		},
+		{
+			context:            "parent address is empty, selected account is not empty, correct password",
+			parentAddress:      "",
+			parentPassword:     TestConfig.Account2.Password,
+			defaultKeyAddress:  TestConfig.Account2.Address,
+			defaultKeyPassword: TestConfig.Account2.Password,
+			expectedError:      nil,
+		},
+	}
+
+	// test
+	for _, testCase := range testCases {
+		s.T().Log(testCase.context)
+
+		// account manager
+		accountManager := account.NewManager(s.NodeManager)
+		require.NotNil(accountManager)
+
+		// select default account if details were provided
+		if len(testCase.defaultKeyAddress) > 0 && len(testCase.defaultKeyPassword) > 0 {
+			require.NoError(accountManager.SelectAccount(testCase.defaultKeyAddress, testCase.defaultKeyPassword))
+		}
+
+		address, pubKey, err := accountManager.CreateChildAccount(testCase.parentAddress, testCase.parentPassword)
+		if !reflect.DeepEqual(err, testCase.expectedError) {
+			s.FailNow(fmt.Sprintf("unexpected error: expected \n'%v', got \n'%v'", testCase.expectedError, err))
+		}
+		if err == nil {
```

Comment:
I think you can avoid using `reflect` here by using `require.EqualValues`. Also, `if err == nil {` check will be more clear if we change it to `if testCase.expectedError == nil {` to emphasize that this is a case "where we expect error to be nil, and run more `NotEmpty` checks on in this case".

So it would be something like:
```go
        require.EqualValues(err, testCase.expectedError)
        if testCase.expectedError == nil {
            require.NotEmpty(address)
            require.NotEmpty(pubKey)
        }
```

Reply:
Reasonable. Legacy should be fixed